### PR TITLE
🎪 Phi2 windows

### PR DESCRIPTION
**Pull request by trajepl**
## Describe your changes
1. Fix phi readme.
    - `python phi2.py cpu_fp32`  -> `python phi2.py --model_type cpu_fp32`
    - onnxruntime-1.17.0 did not support transformer optimization with `phi` model type. It is supported in latest nightly build 1.18.0
2. Provide the windows phi2 optimization with legacy onnx export but not dynamo export which is not supported in windows as of now.

Perf tables:
1. cpu_fp32:
image

2. cpu_int4
image

3. cuda_fp16
image

4. cuda_int4
image

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Lint and apply fixes to your code by running `lintrunner -a`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.

### CODE REVIEW

File: examples/phi2/phi2.py [hunk 1]
```diff
@@ -44,6 +52,14 @@ def main(raw_args=None):
     with open(json_file_template) as f:
         template_json = json.load(f)
 
+    if platform.system() == "Windows":
```
[the remark this answers is not in the document]
Comment (jambayk) [reply examples/phi2/phi2.py:55]: Maybe we should add a note that for the best optimized model, Linux is recommended?

[the remark this answers is not in the document]
Comment (guotuofeng) [reply examples/phi2/phi2.py:55]: for windows, I think the target_opset version need less than 18 I suppose.

File: examples/phi2/phi2.py [hunk 2]
```diff
@@ -35,7 +36,14 @@ def get_args(raw_args):
 
 
 def main(raw_args=None):
-    if version.parse(OrtVersion) < version.parse("1.17.0"):
+    if platform.system() == "Linux" and version.parse(OrtVersion) < version.parse("1.18.0"):
+        raise ValueError(
+            "Please use onnxruntime>=1.18.0 for phi2 optimization in Linux, you can refer to "
```
Comment (guotuofeng) [inline examples/phi2/phi2.py:41]: it need nightly build?

Comment (jambayk) [reply examples/phi2/phi2.py:41]: yes, https://github.com/microsoft/Olive/tree/main/examples/phi2#prerequisites

Comment (trajepl) [reply examples/phi2/phi2.py:41]: Yes, I checked the `phi` model type and found it is only in nightly package. 
For example, in latest 1.17.2, there is not `phi`
https://github.com/microsoft/onnxruntime/blob/rel-1.17.2/onnxruntime/python/tools/transformers/optimizer.py#L45C1-L61C2

**Review by guotuofeng — APPROVED** [review]

**Pull request merged**